main.go

```diff
@@ -93,16 +104,29 @@ func parseFlags() (*config, error) {
 		return nil, fmt.Errorf("unexpected log level: %s", *logLevelRaw)
 	}
 
+	if *rawURL != "" && *rawUpstreamURL != "" {
+		return nil, errors.New("use only one of --url or --upstream.url")
+	}
+
 	if *rawURL != "" {
 		u, err := url.Parse(*rawURL)
 		if err != nil {
 			return nil, err
 		}
 		cfg.url = u
```

Comment:
Maybe we can remove this field entirely and use `cfg.upstreal.url` everywhere now. This would simplify code further down so that we don't have to check both fields everywhere.
```suggestion
```